app/javascript/packs/game.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview This document defines the logic of the game
+ * @requires /api/index.js:MusicApiClient - To fetch the data from db
+ * @requires /model/room.js:Room - Room data which is the basis for the game
+ * @requires /playlist.js:Playlist - Track Audio elements to be played
+ * @requires /loader.js:Loader
+ */
+
+import MusicApiClient from './api';
+import Room from './model/room';
+import Playlist from './playlist';
+import * as LOADER from './loader';
+/** Class representing the game */
+class Game {
+	constructor() {
+		this.room = {};
+		this.playlist = {};
+		this.initialize();
+	}
+	async initialize() {
+		await this.setGameRoomDetails(); // Need to get the game data before adding the event listeners
+		this.addEventListeners();
+	}
+	async setGameRoomDetails() {
+		await this.fetchRoomDetails();
+		this.room = new Room(this.room);
+		this.playlist = new Playlist(this.room.tracksUrl);
```

Comment:
Use the shuffle function when setting the playlist of the room.

```suggestion
              this.playlist = new Playlist(shuffle(this.room.tracksUrl));
```

Reply:
Thanks for the suggestions 😃  
For the shuffling, it would be better to place in the room :)
`this.room`- contains all the data, choices will also be created here
`this.playlist` - contains the view (HTML elements)

In this case, the shuffling would happen in the `this.room` in since the data will be created/instantiated there 😃  If it would be created in the playlist, choices would not be aligned with the playlist being rendered :)

Reply:
Alright, thanks for changing it. Works on my end, you can merge 😊